egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,508 @@
+    keys = utt2spk.keys()
+    keys.sort()
+    for i in range(1, num_replicas+1):
+        for utt_id in keys:
+            new_utt_id = GetNewId(utt_id, prefix, i)
+            corrupted_utt2uniq[new_utt_id] = utt_id
+
+    WriteDictToFile(corrupted_utt2uniq, output_dir + "/utt2uniq")
+
+
+# This function returns only the isotropic noises according to the specified room
+# Please refer to ParseNoiseList() for the format of iso_noise_list
+def FilterIsotropicNoiseList(iso_noise_list, room_id):
+    filtered_list = []
+    for noise in iso_noise_list:
+        if noise.room_linkage == room_id:
+            filtered_list.append(noise)
+            break
+
+    return filtered_list
+
+
+def AddPointSourceNoise(noise_addition_descriptor,  # descriptor to store the information of the noise added
+                        room,  # the room selected
+                        pointsource_noise_list, # the point source noise list
+                        pointsource_noise_addition_probability, # Probability of adding point-source noises
+                        foreground_snrs, # the SNR for adding the foreground noises
+                        background_snrs, # the SNR for adding the background noises
+                        speech_dur,  # duration of the recording
+                        max_noises_recording  # Maximum number of point-source noises that can be added
+                        ):
+    if len(pointsource_noise_list) > 0 and random.random() < pointsource_noise_addition_probability and max_noises_recording > 1:
+        for k in range(random.randint(1, max_noises_recording)):
+            # pick the RIR to reverberate the point-source noise
+            noise = PickItemWithProbability(pointsource_noise_list)
+            noise_rir = PickItemWithProbability(room.rir_list)
+            # If it is a background noise, the noise will be extended and be added to the whole speech
+            # if it is a foreground noise, the noise will not extended and be added at a random time of the speech
+            if noise.bg_fg_type == "background":
+                noise_addition_descriptor['noise_io'].append("wav-reverberate --duration={2} --impulse-response={1} {0} - |".format(noise.noise_file_location, noise_rir.rir_file_location, speech_dur))
+                noise_addition_descriptor['start_times'].append(0)
+                noise_addition_descriptor['snrs'].append(background_snrs.next())
+            else:
+                noise_addition_descriptor['noise_io'].append("wav-reverberate --impulse-response={1} {0} - |".format(noise.noise_file_location, noise_rir.rir_file_location))
+                noise_addition_descriptor['start_times'].append(round(random.random() * speech_dur, 2))
+                noise_addition_descriptor['snrs'].append(foreground_snrs.next())
+
+    return noise_addition_descriptor
+
+
+def GenerateReverberationOpts(room_dict,  # the room dictionary, please refer to MakeRoomDict() for the format
```

Comment:
Please always add a brief description of what the function is supposed to do. Please remember what your code has to be easy to read and maintain.